packages/nimble-components/src/select/index.ts

```diff
@@ -668,6 +673,17 @@ export class Select extends FormAssociatedSelect implements ErrorPattern {
         }
     }
 
+    /**
+     * @internal
+     */
+    public registerOption(option: ListOption): void {
+        if (this.options.includes(option)) {
+            return;
+        }
+
+        this.options.push(option);
```

Comment:
For the tech debt item that is created about synchronous state can you link to that from here as well? And discuss in the issue that there can be state inaccuracies if options are inserted in a different order than just appending.